site-src/api-types/grpcroute.md

```diff
@@ -0,0 +1,216 @@
+based on feedback during the alpha stage.
+
+Conformance levels are defined by the filter type:
+
+ - All "core" filters MUST be supported by implementations supporting GRPCRoute.
+ - Implementers are encouraged to support "extended" filters.
+ - "Custom" filters have no API guarantees across implementations.
+
+Specifying a core filter multiple times has unspecified or custom conformance.
+
+All filters are expected to be compatible with each other. If an implementation
+cannot support other combinations of filters, they must clearly document that
+limitation. In all cases where incompatible or unsupported filters are
+specified, implementations MUST add a warning condition to status.
+
+#### BackendRefs (optional)
+
+BackendRefs defines the API objects to which matching requests should be sent. If
+unspecified, the rule performs no forwarding. If unspecified and no filters
+are specified that would result in a response being sent, an `UNIMPLEMENTED` error code
+is returned.
+
+The following example forwards gRPC requests for the method `User.Login` to service
+"my-service1" on port `50051` and gRPC requests for the method `Things.DoThing` with
+header `magic: foo` to service "my-service2" on port `50051`:
+```yaml
+{% include 'experimental/basic-grpc.yaml' %}
+```
+
+The following example uses the `weight` field to forward 90% of gRPC requests to
+`foo.example.com` to the "foo-v1" Service and the other 10% to the "foo-v2"
+Service:
+```yaml
+{% include 'experimental/traffic-splitting/grpc-traffic-split-2.yaml' %}
+```
+
+Reference the [backendRef][backendRef] API documentation for additional details
+on `weight` and other fields.
+
+## Status
+
+Status defines the observed state of the GRPCRoute.
+
+### RouteStatus
+
+RouteStatus defines the observed state that is required across all route types.
+
+#### Parents
+
+Parents define a list of the Gateways (or other parent resources) that are
+associated with the GRPCRoute, and the status of the GRPCRoute with respect to
+each of these Gateways. When a GRPCRoute adds a reference to a Gateway in
+parentRefs, the controller that manages the Gateway should add an entry to this
+list when the controller first sees the route and should update the entry as
+appropriate when the route is modified.
+
+The following example indicates GRPCRoute "grpc-example" has been accepted by
+Gateway "gw-example" in namespace "gw-example-ns":
+```yaml
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: GRPCRoute
+metadata:
+  name: grpc-example
+...
+status:
+  parents:
+  - parentRefs:
+      name: gw-example
+      namespace: gw-example-ns
+    conditions:
+    - type: Accepted
+      status: "True"
+```
+
+## Merging
+Multiple GRPCRoutes can be attached to a single Gateway resource. Importantly,
+only one Route rule may match each request. For more information on how conflict
+resolution applies to merging, refer to the [API specification][grpcrouterule].
```

Comment:
I think it would be helpful in this doc to answer the following questions:

1. Why does this resource exist instead of just reusing HTTPRoute?
2. How does merging work when both HTTPRoutes and GRPCRoutes are attached?